service/history/decisionHandler.go

```diff
@@ -655,12 +655,20 @@ func (handler *decisionHandlerImpl) handleBufferedQueries(
 	// Consistent query requires both server and client worker support. If a consistent query was requested (meaning there are
 	// buffered queries) but worker does not support consistent query then all buffered queries should be failed.
 	if versionErr := handler.versionChecker.SupportsConsistentQuery(clientImpl, clientFeatureVersion); versionErr != nil {
+		scope.IncCounter(metrics.WorkerNotSupportsConsistentQueryCount)
 		failedTerminationState := &queryTerminationState{
 			queryTerminationType: queryTerminationTypeFailed,
 			failure:              versionErr,
 		}
 		buffered := queryRegistry.getBufferedIDs()
 		for _, id := range buffered {
+			handler.logger.Info(
+				"failing query because worker does not support consistent query",
+				tag.WorkflowDomainName(domain),
+				tag.WorkflowID(workflowID),
+				tag.WorkflowRunID(runID),
+				tag.QueryID(id),
```

Comment:
Are these ids human readable or just uuids? If it's the latter we can remove it and move the log outside of the for loop.